### PR TITLE
bazel-buildtools: 0.22.0 -> 0.28.0

### PR DESCRIPTION
**Pull request by u-quark**
###### Motivation for this change
Continuation from #64537

Update to the latest upstream version release: 0.28.0.

I've updated the dependencies by using go2nix. The latest version of go2nix changed a bit the derivation (adding `fetchhg`, `fetchbzr`, `fetchsvn` even if they are not used).

cc @Profpatsch @uri-canva 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/build-managers/bazel/buildtools/default.nix [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 
-  src = fetchFromGitHub {
-    owner = "bazelbuild";
-    repo = "buildtools";
-    rev = "55b64c3d2ddfb57f06477c1d94ef477419c96bd6";
-    sha256 = "0n6q8pkgy3vvmwyrxvkmjfbcxc31i31czg2bjdzq7awwrr4fdbwy";
+  src = fetchgit {
+    inherit rev;
+    url = "https://github.com/bazelbuild/buildtools";
+    sha256 = "1d8zjgbg77sk27cz9pjz1h6ajwxqmvdzqgwa2jbh6iykibhpadq0";
```
Comment (Profpatsch) [inline pkgs/development/tools/build-managers/bazel/buildtools/default.nix]: Any reason to replace `fetchFromGitHub` here?

Comment (u-quark) [reply pkgs/development/tools/build-managers/bazel/buildtools/default.nix]: @Profpatsch  I am trying to keep as much of the derivation returned from go2nix.

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build bazel-buildtools

**Comment by Profpatsch** [comment]
CI is failing

**Comment by u-quark** [comment]
I will play around and see I can make CI succeed.

**Comment by u-quark** [comment]
@Profpatsch Is CI working now? I added back the `github.com/google/skylark` as a dependency, that was what was failing before. Not sure why go2nix did not pick this up. This is the latest version of that library (it has been renamed to https://github.com/google/starlark-go but the code still references `github.com/google/skylark`).

**Comment by Profpatsch** [comment]
@GrahamcOfBorg build bazel-buildtools

**Comment by u-quark** [comment]
Thanks @Profpatsch

**Pull request merged**